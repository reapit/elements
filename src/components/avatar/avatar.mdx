import { Meta, Story, Canvas, Controls } from '@storybook/blocks'
import { RenderHtmlMarkup } from '../../storybook/render-html-markup'
import * as AvatarStories from './avatar.stories'

<Meta of={AvatarStories} />

# Avatar

An avatar component to be used typically with a `Card` or `Nav` component. The default variant renders a circle with children.

There are multiple variants that can be used by utilizing the data props, such as size, colour, and shape (see the HTML version of React usage). In addition to text, it also accepts an Icon as a child.

## Style Only Usage

<Canvas of={AvatarStories.StyleOnlyUsage} />

<RenderHtmlMarkup of={AvatarStories.StyleOnlyUsage} />

<Canvas of={AvatarStories.WithIconUsage} />

<RenderHtmlMarkup of={AvatarStories.WithIconUsage} />

## With Colour

<Canvas of={AvatarStories.WithColour} />

<RenderHtmlMarkup of={AvatarStories.WithColour} />

## With Square Shape

<Canvas of={AvatarStories.WithSquareShape} />

<RenderHtmlMarkup of={AvatarStories.WithSquareShape} />

## With Small Size

<Canvas of={AvatarStories.WithSmallSize} />

<RenderHtmlMarkup of={AvatarStories.WithSmallSize} />

## React Usage

<Controls />

<Canvas of={AvatarStories.ReactUsage} />

<RenderHtmlMarkup of={AvatarStories.ReactUsage} />WithColourWithColour
