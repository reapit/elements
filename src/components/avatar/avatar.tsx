import React, { FC, HTMLAttributes } from 'react'
import { cx } from '@linaria/core'
import { ElAvatar, ElAvatarImage } from './__styles__'

export interface AvatarProps extends HTMLAttributes<HTMLSpanElement> {
  type?: 'profile' | 'image'
  src?: string
  alt?: string
}

export const Avatar: FC<AvatarProps> = ({ children, src, alt, type, className, ...rest }) => {
  return type === 'image' ? (
    <ElAvatarImage role="img" title="An avatar image" className={cx(className)} {...rest}>
      {src ? <img src={src} alt={alt ? alt : `An image with source ${src}`} /> : children}
    </ElAvatarImage>
  ) : (
    <ElAvatar role="img" title="A profile image" className={cx(className)} {...rest}>
      {src ? <img src={src} alt={alt ? alt : `An image with source ${src}`} /> : children}
    </ElAvatar>
  )
}
