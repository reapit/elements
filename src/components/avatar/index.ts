export * from './styles'
export * from './avatar'
