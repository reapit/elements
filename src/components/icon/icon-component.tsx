import { FC, HTMLAttributes, memo } from 'react'
import { cx } from '@linaria/core'
import { Intent, getIntentClassName } from '../../helpers/intent'
import { ElIcon } from './__styles__'
import { iconSet } from '../../icons'
export type IconNames = keyof typeof iconSet

export interface IconProps extends HTMLAttributes<HTMLSpanElement> {
  icon: IconNames
  intent?: Intent
  fontSize?: string
  className?: string
  height?: string
  width?: string
}

export const Icon: FC<IconProps> = memo(({ icon, intent, fontSize, className, height, width, ...rest }) => {
  const Svg = iconSet[icon]
  const intentClassname = intent && getIntentClassName(intent)
  const combinedClassName = cx(className, intentClassname)

  if (!Svg) return <ElIcon className={combinedClassName} {...rest} />

  return (
    <ElIcon className={combinedClassName} {...rest} style={{ fontSize, height, width }}>
      <Svg role="img" title={`Icon image with name ${icon}`} style={{ fontSize, height, width }} />
    </ElIcon>
  )
})
