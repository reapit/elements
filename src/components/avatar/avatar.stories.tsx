import type { Meta, StoryObj } from '@storybook/react'
import { Avatar } from '.'
import { Icon } from '../icon'

export default {
  title: 'Components/Avatar',
  component: Avatar,
  args: {
    children: 'AD',
  },
} as Meta<typeof Avatar>

type Story = StoryObj<typeof Avatar>

/**
 * An avatar component to be used typically with a `Card` or `Nav` component.
 * The default variant renders a circle with children.
 *
 * There are multiple variants that can be used by utilizing the `data-` props
 * such as `size`, `colour`, and `shape` (see the HTML version of each version).
 * In addition to text, it also accepts an Icon as a child.
 */
export const Default: Story = {}

export const AvatarWithIcon: Story = {
  render: (props) => (
    <Avatar {...props}>
      <Icon icon="contact" />
    </Avatar>
  ),
}

export const AvatarColour: Story = {
  args: {
    colour: 'purple',
  },
}

export const AvatarShape: Story = {
  args: {
    shape: 'square',
  },
}

export const AvatarSize: Story = {
  args: {
    size: 'small',
  },
}
