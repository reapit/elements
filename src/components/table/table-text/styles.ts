import { styled } from '@linaria/react'
import { ElIcon } from '../../icon'

export const ElTableTextIcon = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const ElTableTextContent = styled.span`
  display: flex;
  align-items: center;
  gap: 10px;
  color: var(--colour-text-primary);
  font-family: var(--font-sm-regular-family);
  font-style: normal;
  font-weight: var(--font-sm-regular-weight);
  font-size: var(--font-sm-regular-size);
  line-height: var(--font-sm-regular-line_height);
  letter-spacing: var(--font-sm-regular-letter_spacing);
`

export const ElTableText = styled.div`
  display: flex;
  align-items: center;
  gap: var(--spacing-2);

  ${ElTableTextIcon} {
    color: var(--colour-icon-primary);
    ${ElIcon} {
      width: var(--icon_size-s);
      height: var(--icon_size-s);
      color: inherit;
    }
  }

  &[data-size='extra-small'] {
    ${ElTableTextContent} {
      font-size: var(--font-xs-regular-size);
      line-height: var(--font-xs-regular-line_height);
      letter-spacing: var(--font-xs-regular-letter_spacing);
      color: var(--colour-text-secondary);
    }
  }

  &[data-variant='primary'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
    }
  }

  &[data-variant='secondary'] {
    ${ElTableTextIcon} {
      color: var(--icon-secondary);
    }
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-secondary);
    }
  }

  &[data-variant='info'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-info);
    }
    ${ElTableTextIcon} {
      color: var(--icon-info);
    }
  }

  &[data-variant='success'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-success);
    }
    ${ElTableTextIcon} {
      color: var(--icon-success);
    }
  }

  &[data-variant='pending'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-pending);
    }
    ${ElTableTextIcon} {
      color: var(--icon-pending);
    }
  }

  &[data-variant='warning'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-warning);
    }
    ${ElTableTextIcon} {
      color: var(--icon-warning);
    }
  }

  &[data-variant='error'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-error);
    }
    ${ElTableTextIcon} {
      color: var(--icon-error);
    }
  }

  &[data-variant='accent_1'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-accent_1);
    }
    ${ElTableTextIcon} {
      color: var(--icon-accent_1);
    }
  }

  &[data-variant='accent_2'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-accent_2);
    }
    ${ElTableTextIcon} {
      color: var(--icon-accent_2);
    }
  }

  &[data-variant='action'] {
    ${ElTableTextContent} {
      font-weight: var(--font-sm-medium-weight);
      line-height: var(--font-sm-medium-line_height);
      letter-spacing: var(--font-sm-medium-letter_spacing);
      color: var(--colour-text-action);
    }
    ${ElTableTextIcon} {
      color: var(--icon-action);
    }
  }
`
