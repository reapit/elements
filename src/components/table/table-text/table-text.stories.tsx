import { TableText } from './table-text'
import { Icon, IconNames } from '../../icon'

import type { Meta, StoryObj } from '@storybook/react'

const ICON_OPTIONS: IconNames[] = ['star', 'add', 'chevronDown', 'chevronLeft', 'chevronRight']

const meta = {
  title: 'Components/TableText',
  component: TableText,
  argTypes: {
    children: {
      control: 'text',
      description: 'Defines the text for the Table Text component.',
    },
    variant: {
      control: 'select',
      options: [
        'primary',
        'secondary',
        'info',
        'success',
        'pending',
        'warning',
        'error',
        'accent_1',
        'accent_2',
        'action',
      ],
      description: 'Defines the Table Text style variant.',
    },
    size: {
      control: 'select',
      options: ['small', 'extra-small'],
      description: 'Defines the Table Text size.',
    },
    iconLeft: {
      control: 'select',
      options: ICON_OPTIONS,
      description: 'The icon displayed on the left side of the badge.',
      mapping: ICON_OPTIONS.reduce((acc, iconName) => {
        acc[iconName] = <Icon icon={iconName} fontSize="1rem" />
        return acc
      }, {}),
    },
    iconRight: {
      control: 'select',
      options: ICON_OPTIONS,
      description: 'The icon displayed on the right side of the badge.',
      mapping: ICON_OPTIONS.reduce((acc, iconName) => {
        acc[iconName] = <Icon icon={iconName} fontSize="1rem" />
        return acc
      }, {}),
    },
  },
} satisfies Meta<typeof TableText>

export default meta

type Story = StoryObj<typeof meta>

/**
 * The table text component is designed to be used with in the table component.
 * It can have different color varients and can be of only two sizes 'small' or 'extra-small'
 */
export const BasicUsage: Story = {
  args: {
    children: 'Value',
  },
}

/**
 * The table text component can be of only two sizes 'small' or 'extra-small'. Default varient will be 'secondary'
 */
export const TableTextSize: Story = {
  args: {
    ...BasicUsage.args,
    size: 'extra-small',
  },
}

/**
 * The table text component can have different color varients 'primary', 'secondary',
 * 'info', 'success', 'pending', 'warning', 'error', 'accent_1', 'accent_2', 'action',
 */
export const TableTextVarients: Story = {
  args: {
    ...BasicUsage.args,
  },
  // Adding decorator into the story to display multiple stories.
  decorators: [
    (Story: any) => (
      <div
        style={{
          display: 'flex',
          gap: '10px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  render: (props: any) => (
    <>
      <TableText {...props}>{props.children}</TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'secondary'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'info'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'success'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'pending'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'error'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'accent_1'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'accent_2'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'action'}>
        {props.children}
      </TableText>
      <TableText {...props} variant={props.variant ? props.variant : 'primary'}>
        {props.children}
      </TableText>
    </>
  ),
  // Adding the parameters to display only one varient code in the source
  parameters: {
    docs: {
      source: {
        code: `
  <TableText variant='success'>
  Value
</TableText>
          `,
      },
    },
  },
}

/**
 * The table text component can icon on the left or right or just icon only.
 */
export const TableTextWithIcon: Story = {
  args: {
    ...BasicUsage.args,
    iconLeft: 'star',
    iconRight: 'star',
  },
}
