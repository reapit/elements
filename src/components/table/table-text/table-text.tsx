import React, { ReactNode, HTMLAttributes } from 'react'
import { ElTableText, ElTableTextIcon, ElTableTextContainer, ElTableTextContent } from './styles'

type TableTextVariant =
  | 'primary'
  | 'secondary'
  | 'info'
  | 'success'
  | 'pending'
  | 'warning'
  | 'error'
  | 'accent_1'
  | 'accent_2'
  | 'action'
type TableTextSize = 'small' | 'extra-small'

export interface TableTextProps extends HTMLAttributes<HTMLDivElement> {
  variant?: TableTextVariant
  size?: TableTextSize
  iconLeft?: ReactNode
  iconRight?: ReactNode
}

export const TableText: React.FC<TableTextProps> = ({ children, variant, size, iconLeft, iconRight, ...rest }) => {
  return (
    <ElTableText data-variant={variant} data-size={size} {...rest}>
      {iconLeft && <ElTableTextIcon>{iconLeft}</ElTableTextIcon>}
      {children && (
        <ElTableTextContainer>
          <ElTableTextContent>{children}</ElTableTextContent>
        </ElTableTextContainer>
      )}
      {iconRight && <ElTableTextIcon>{iconRight}</ElTableTextIcon>}
    </ElTableText>
  )
}
