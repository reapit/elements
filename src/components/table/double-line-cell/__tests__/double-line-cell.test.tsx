import { render } from '@testing-library/react'
import { DoubleLineCell } from '..'
import { TableText } from '../../table-text'
import { DeprecatedIcon } from '#src/components/deprecated-icon/icon-component'
import { Avatar } from '#src/components/avatar/avatar'

describe('DoubleLineCell', () => {
  test('should match snapshot', () => {
    const { asFragment } = render(<DoubleLineCell />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot with first and second line', () => {
    const { asFragment } = render(
      <DoubleLineCell
        firstLine={<TableText>Alphanumeric value</TableText>}
        secondLine={<TableText size="extra-small">Value 23 Jan 2025 4:30 pm Value</TableText>}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot with first and second line with icons', () => {
    const { asFragment } = render(
      <DoubleLineCell
        firstLine={
          <>
            <DeprecatedIcon icon="star" fontSize="1rem" />
            <TableText>Alphanumeric value</TableText>
            <DeprecatedIcon icon="star" fontSize="1rem" />
          </>
        }
        secondLine={<TableText size="extra-small">Value 23 Jan 2025 4:30 pm Value</TableText>}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should match snapshot with first and second line with media item', () => {
    const { asFragment } = render(
      <DoubleLineCell
        mediaItem={<Avatar>JS</Avatar>}
        firstLine={<TableText>John Smith</TableText>}
        secondLine={<TableText size="extra-small">Plumber</TableText>}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
