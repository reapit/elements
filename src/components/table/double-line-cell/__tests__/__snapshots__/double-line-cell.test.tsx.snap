// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DoubleLineCell > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-double-line-cell"
  >
    <div
      class="mocked-styled-2 el-cell-content"
    >
      <div
        class="mocked-styled-5 el-first-line"
      />
      <div
        class="mocked-styled-6 el-second-line"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DoubleLineCell > should match snapshot with first and second line 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-double-line-cell"
  >
    <div
      class="mocked-styled-2 el-cell-content"
    >
      <div
        class="mocked-styled-5 el-first-line"
      >
        <div
          class="mocked-styled-9 el-table-text"
        >
          <div
            class="mocked-styled-10 el-table-text-container"
          >
            <span
              class="mocked-styled-8 el-table-text-content"
            >
              Alphanumeric value
            </span>
          </div>
        </div>
      </div>
      <div
        class="mocked-styled-6 el-second-line"
      >
        <div
          class="mocked-styled-9 el-table-text"
          data-size="extra-small"
        >
          <div
            class="mocked-styled-10 el-table-text-container"
          >
            <span
              class="mocked-styled-8 el-table-text-content"
            >
              Value 23 Jan 2025 4:30 pm Value
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DoubleLineCell > should match snapshot with first and second line with icons 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-double-line-cell"
  >
    <div
      class="mocked-styled-2 el-cell-content"
    >
      <div
        class="mocked-styled-5 el-first-line"
      >
        <span
          class="mocked-styled-11 el-icon"
          style="font-size: 1rem;"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            style="font-size: 1rem;"
            title="Icon image with name star"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.2341 13.9495L21.8141 9.8795C22.0176 9.76139 22.1112 9.51746 22.0389 9.2936C21.9667 9.06973 21.7482 8.92648 21.5141 8.9495H14.6241L12.4941 2.4095C12.4463 2.17134 12.2371 2 11.9941 2C11.7512 2 11.542 2.17134 11.4941 2.4095L9.40414 8.9495H2.51414C2.2932 8.94327 2.09438 9.08284 2.02518 9.29275C1.95597 9.50267 2.03281 9.73312 2.21414 9.8595L7.79414 13.9495L5.66414 20.4895C5.59621 20.6958 5.66929 20.9224 5.84498 21.0502C6.02067 21.178 6.25876 21.1777 6.43414 21.0495L12.0141 16.9495L17.5941 21.0095C17.7695 21.1377 18.0076 21.138 18.1833 21.0102C18.359 20.8824 18.4321 20.6558 18.3641 20.4495L16.2341 13.9495Z"
              fill="currentColor"
            />
          </svg>
        </span>
        <div
          class="mocked-styled-9 el-table-text"
        >
          <div
            class="mocked-styled-10 el-table-text-container"
          >
            <span
              class="mocked-styled-8 el-table-text-content"
            >
              Alphanumeric value
            </span>
          </div>
        </div>
        <span
          class="mocked-styled-11 el-icon"
          style="font-size: 1rem;"
        >
          <svg
            fill="none"
            height="1em"
            role="img"
            style="font-size: 1rem;"
            title="Icon image with name star"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.2341 13.9495L21.8141 9.8795C22.0176 9.76139 22.1112 9.51746 22.0389 9.2936C21.9667 9.06973 21.7482 8.92648 21.5141 8.9495H14.6241L12.4941 2.4095C12.4463 2.17134 12.2371 2 11.9941 2C11.7512 2 11.542 2.17134 11.4941 2.4095L9.40414 8.9495H2.51414C2.2932 8.94327 2.09438 9.08284 2.02518 9.29275C1.95597 9.50267 2.03281 9.73312 2.21414 9.8595L7.79414 13.9495L5.66414 20.4895C5.59621 20.6958 5.66929 20.9224 5.84498 21.0502C6.02067 21.178 6.25876 21.1777 6.43414 21.0495L12.0141 16.9495L17.5941 21.0095C17.7695 21.1377 18.0076 21.138 18.1833 21.0102C18.359 20.8824 18.4321 20.6558 18.3641 20.4495L16.2341 13.9495Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <div
        class="mocked-styled-6 el-second-line"
      >
        <div
          class="mocked-styled-9 el-table-text"
          data-size="extra-small"
        >
          <div
            class="mocked-styled-10 el-table-text-container"
          >
            <span
              class="mocked-styled-8 el-table-text-content"
            >
              Value 23 Jan 2025 4:30 pm Value
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DoubleLineCell > should match snapshot with first and second line with media item 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-0 el-double-line-cell"
  >
    <div
      class="mocked-styled-1 el-avatar-content"
    >
      <div
        class="mocked-styled-3 el-media-item-content"
      >
        <span
          class="mocked-styled-12 el-avatar"
          data-colour="default"
          data-shape="circle"
          data-size="medium"
          role="presentation"
        >
          JS
        </span>
      </div>
      <div
        class="mocked-styled-4 el-double-line-content"
      >
        <div
          class="mocked-styled-5 el-first-line"
        >
          <div
            class="mocked-styled-9 el-table-text"
          >
            <div
              class="mocked-styled-10 el-table-text-container"
            >
              <span
                class="mocked-styled-8 el-table-text-content"
              >
                John Smith
              </span>
            </div>
          </div>
        </div>
        <div
          class="mocked-styled-6 el-second-line"
        >
          <div
            class="mocked-styled-9 el-table-text"
            data-size="extra-small"
          >
            <div
              class="mocked-styled-10 el-table-text-container"
            >
              <span
                class="mocked-styled-8 el-table-text-content"
              >
                Plumber
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
